test/auth.test.tsx

```diff
@@ -123,6 +127,67 @@ describe('Authentication', () => {
     test.todo('checks requiredClaims');
   });
 
+  describe('useSigninCheck()', () => {
+    it('accurately reflects signed-in state', async () => {
+      const { result, waitFor: waitForHookCondition } = renderHook(() => useSigninCheck(), { wrapper: Provider });
+
+      await waitForHookCondition(() => result.current.status === 'success');
+
+      // Signed out
+      expect(app.auth().currentUser).toBeNull();
+      expect(result.current.data).toEqual({ signedIn: false, hasRequiredClaims: false });
+
+      await hooksAct(async () => {
+        await signIn();
+      });
+
+      // Signed in
+      expect(app.auth().currentUser).not.toBeNull();
+      expect(result.current.data).toEqual({ signedIn: true, hasRequiredClaims: true, user: app.auth().currentUser });
+    });
+
+    // Skipping this because of an issue setting custom claims
+    // https://github.com/firebase/firebase-tools/issues/3083
+    it.skip('checks custom claims', async () => {
```

Comment:
skipping this test because of https://github.com/firebase/firebase-tools/issues/3083

Reply:
fixed this